### PR TITLE
RFC: Fix `line_z` for PyPlot backend (fix #1021)

### LINKED ISSUE #1021: pyplot(): zcolor keyword not working with lines
I can't make the colorbar to appear with the `plot` command:

`using Plots; pyplot()`
`plot(rand(10,6), zcolor=1:10, colorbar=true, leg=false)`
wrong

However, when calling `scatter` it does somehow works (https://github.com/JuliaPlots/Plots.jl/issues/412).

### PR DESCRIPTION
**Pull request by darwindarak**
```julia
s = linspace(0, 10, 50)
plot(s.*cos.(s), s.*sin.(s), s, line_z = s, lw = 3)
```
3d_linez
```julia
n = 10; x = linspace(0,1,n); y=cumsum(rand(n));
plot(x,y,line_z=1:n, w=5)
```
2d_linez

### CODE REVIEW

**Comment by darwindarak** [comment]
Also normalize colors when hiding colorbars.
```julia
plot([1:10, 1:10], [1:10 1:10], line_z = 0:0.01:0.1, lw = 4,
    clim=(-0.03, 0.03), layout = (2,1), cbar = [:none :right])
```
Before:
normalize_colors

After:
normalize_colors_after

**Comment by darwindarak** [comment]
Hmm, nothing shows up right now when you run that.  I'll take a closer look at that this weekend.

File: src/backends/pyplot.jl [hunk 1]
```diff
@@ -499,32 +499,31 @@ function py_add_series(plt::Plot{PyPlotBackend}, series::Series)
...
                     :cmap => py_linecolormap(series),
                     :linewidth => py_dpi_scale(plt, series[:linewidth]),
-                    :linestyle => py_linestyle(st, series[:linestyle])
+                    :linestyle => py_linestyle(st, series[:linestyle]),
+                    :norm => pycolors["Normalize"](; extrakw...)
                 )
-                if needs_colorbar
-                    kw[:norm] = pycolors["Normalize"](; extrakw...)
-                end
                 lz = collect(series[:line_z])
```
Comment (daschw) [inline src/backends/pyplot.jl]: Could you try to replace this line with
```julia
lz = _cycle(series[:line_z], 1:n)
```
?

Comment (darwindarak) [reply src/backends/pyplot.jl]: Nice, that did it

multicolored

**Comment by daschw** [comment]
I will merge this now. Thanks a lot @darwindarak!
If you're up for it you could also look into the legend line colors with line_z ;-) That would be great!
For GR I chose the color of the first segment to be displayed in the legend.

**Pull request merged**